src/librustc_mir/interpret/memory.rs

```diff
@@ -707,10 +707,10 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
                 new_relocations.extend(
                     relocations
                     .iter()
-                    .map(|&(offset, reloc)| {
-                    (offset + dest.offset - src.offset + (i * size * relocations.len() as u64),
-                     reloc)
-                    })
+                    .map(|&(offset, reloc)| (
+                        offset + dest.offset - src.offset + (i * size),
```

Comment:
Might be clearer to do something like `let dest_offset = dest.offset + (i * size);` and then `offset + dest_offset - src.offset` to show that the `i * size` is part of the destination offset (I started at this for a bit wondering where it is coming from).